tests/Client/UpdaterTest.php

```diff
@@ -97,123 +125,257 @@ public function providerRefreshRepository()
+                    'targets' => 3,
                 ],
             ],
         ], 0);
     }
 
     /**
-     * Asserts that files in the repo are at expected versions.
+     * Asserts that files in the client repo are at expected versions.
      *
      * @param array $expectedVersions
      *   The expected versions.
      *
      * @return void
      */
-    protected function assertRepoVersions(array $expectedVersions): void
+    protected function assertClientRepoVersions(array $expectedVersions): void
     {
-        $this->assertSame(
-            $expectedVersions['root'],
-            RootMetadata::createFromJson($this->localRepo['root.json'])
-            ->getVersion()
-        );
-        $this->assertSame(
-            $expectedVersions['timestamp'],
-            TimestampMetadata::createFromJson($this->localRepo['timestamp.json'])
-            ->getVersion()
-        );
-        $this->assertSame(
-            $expectedVersions['snapshot'],
-            SnapshotMetadata::createFromJson($this->localRepo['snapshot.json'])
-            ->getVersion()
-        );
+        foreach ($expectedVersions as $type => $version) {
+            if (is_null($version)) {
+                $this->assertNull($this->localRepo["$type.json"]);
+                return;
+            }
+            $actualVersion = MetadataBase::createFromJson($this->localRepo["$type.json"])->getVersion();
+            $this->assertSame(
+                $expectedVersions[$type],
+                $actualVersion,
+                "Actual version of $type, '$actualVersion' does not match expected version '$version'"
+            );
+        }
     }
 
     /**
-     * Tests that an exception is thrown when attempting to refresh the repository with file that has an invalid valid
-     * signature.
+     * Tests that exceptions are thrown when metadata files are not valid.
      *
-     * @param string $fileToFail
-     *   The repo fail that should fail the signature check.
-     * @param integer $expectedRootVersion
-     *   The expected root version after the refresh attempt.
-     * @param string $expectionMessage
-     *   The expected exception message.
+     * @param string $fileToChange
+     *   The file to change.
+     * @param array $keys
+     *   The nested keys of the element to change.
+     * @param mixed $newValue
+     *   The new value to set.
+     * @param \Exception $expectionException
+     *   The excpected exception.
+     * @param array $expectedUpdatedVersions
+     *   The expected repo file version after refresh attempt.
      *
      * @return void
      *
-     * @dataProvider providerSignatureError
+     * @dataProvider providerRefreshException
      */
-    public function testSignatureError(string $fileToFail, int $expectedRootVersion, string $expectionMessage) : void
+    public function testRefreshException(string $fileToChange, array $keys, $newValue, \Exception $expectionException, array $expectedUpdatedVersions): void
```

Comment:
Instead of a specific test for signature errors I changed this into generic method to test different types of exceptions when files are changed.